Remove PlatformMediaSessionManager::singleton

eric-carlson opened the pull request:
#### 17897423938e86e276fab1d57b6ef0d99944e41d
<pre>
Remove PlatformMediaSessionManager::singleton
<a href="https://bugs.webkit.org/show_bug.cgi?id=292687">https://bugs.webkit.org/show_bug.cgi?id=292687</a>
<a href="https://rdar.apple.com/150885431">rdar://150885431</a>

Reviewed by Andy Estes.

For site-isolation we will need to move MediaSessionManager and related classes to the
GPU process and have a different MediaSessionManagerInterface proxy for every Page. As a
step in this direction, get rid of PlatformMediaSessionManager::singleton and add
Page::mediaSessionManager and remove all PlatformMediaSessionManager static methods.

Add a factory function to PageConfiguration that WebProcess will use to create a
RemoveMediaSessionManager once it exists.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/audiosession/DOMAudioSession.cpp:
(WebCore::DOMAudioSession::setType): Use Page::mediaSessionManager instead of
PlatformMediaSessionManager::singleton.

* Source/WebCore/Modules/mediacapabilities/MediaCapabilities.cpp:
(WebCore::gatherDecodingInfo): Ditto.

* Source/WebCore/Modules/mediasession/MediaSession.cpp:
(WebCore::MediaSession::setActionHandler): Ditto.
(WebCore::MediaSession::protectedDocument const): Add Document RefPtr
(WebCore::MediaSession::sessionManager const): Add accessor.
(WebCore::MediaSession::activeMediaElement const):
(WebCore::MediaSession::updateNowPlayingInfo):
(WebCore::MediaSession::updateCaptureState):
(WebCore::MediaSession::document const): Deleted.
* Source/WebCore/Modules/mediasession/MediaSession.h:

* Source/WebCore/Modules/mediastream/MediaStream.cpp:
(WebCore::MediaStream::startProducingData): Call `sessionCanProduceAudioChanged` if the
stream has an audio track so we don&apos;t need to do call it from the Mock audio source.
(WebCore::MediaStream::mediaSessionManager const):
* Source/WebCore/Modules/mediastream/MediaStream.h:

* Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp:
(WebCore::MediaStreamTrack::MediaStreamTrack): Use Page::mediaSessionManager instead of
PlatformMediaSessionManager::singleton.
(WebCore::MediaStreamTrack::~MediaStreamTrack): Ditto.
(WebCore::MediaStreamTrack::stopTrack): Ditto.
(WebCore::MediaStreamTrack::trackEnded): Ditto.
(WebCore::MediaStreamTrack::trackMutedChanged): Ditto.
(WebCore::MediaStreamTrack::mediaSessionManager const):
* Source/WebCore/Modules/mediastream/MediaStreamTrack.h:

* Source/WebCore/Modules/mediastream/UserMediaRequest.cpp:
(WebCore::UserMediaRequest::start): Ditto.

* Source/WebCore/Modules/webaudio/AudioContext.cpp:
(WebCore::AudioContext::defaultDestinationWillBecomeConnected): Ditto.
(WebCore::AudioContext::sessionManager const): Ditto.
* Source/WebCore/Modules/webaudio/AudioContext.h:

* Source/WebCore/Modules/webaudio/BaseAudioContext.cpp:
(WebCore::BaseAudioContext::setState): Ditto.
(WebCore::BaseAudioContext::mediaSessionManager const):
* Source/WebCore/Modules/webaudio/BaseAudioContext.h:

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::visibilityStateChanged): Ditto.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::bestMediaElementForRemoteControls): Deleted.
(WebCore::HTMLMediaElement::seekToPlaybackPositionEndedTimerFired): Use
Page::mediaSessionManager instead of PlatformMediaSessionManager::singleton.
(WebCore::HTMLMediaElement::couldPlayIfEnoughData const): Ditto.
(WebCore::HTMLMediaElement::shouldDisableSleep const): Ditto.
(WebCore::HTMLMediaElement::shouldOverrideBackgroundPlaybackRestriction const): Ditto.
(WebCore::HTMLMediaElement::sessionManager const):
* Source/WebCore/html/HTMLMediaElement.h:

* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::MediaElementSession::isVisibleInViewportChanged): Use
PlatformMediaSessionClient::sessionManager instead of PlatformMediaSessionManager::singleton.
(WebCore::MediaElementSession::clientDataBufferingTimerFired): Ditto.
(WebCore::MediaElementSession::updateClientDataBuffering): Ditto.
(WebCore::MediaElementSession::canShowControlsManager const): Ditto.
(WebCore::MediaElementSession::setHasPlaybackTargetAvailabilityListeners): Ditto.

* Source/WebCore/page/Page.cpp:
(WebCore::Page::bestMediaElementForRemoteControls): Move from HTMLMediaElement.
(WebCore::Page::mediaPlaybackExists): Use mediaSessionManager instance variable instead of
PlatformMediaSessionManager::singletonIfExists.
(WebCore::Page::mediaPlaybackIsPaused): Ditto.
(WebCore::Page::pauseAllMediaPlayback): Ditto.
(WebCore::Page::suspendAllMediaPlayback): Ditto.
(WebCore::Page::resumeAllMediaPlayback): Ditto.
(WebCore::Page::suspendAllMediaBuffering): Ditto.
(WebCore::Page::resumeAllMediaBuffering): Ditto.
(WebCore::Page::setActivityState): Ditto.
(WebCore::Page::activeNowPlayingSessionUpdateTimerFired): Ditto.
(WebCore::Page::setPresentingApplicationAuditToken): Ditto.
(WebCore::Page::mediaSessionManager): Allocate `m_mediaSessionManager` with the factory
function if it doesn&apos;t exist. If m_mediaSessionManagerFactory is null, true for WK1 and
WK2 until we move the session manager to the GPU process, create a factory that allocates
and instance of PlatformMediaSessionManager.
(WebCore::Page::mediaSessionManagerIfExists const):
(WebCore::Page::mediaSessionManagerForPageIdentifier):
* Source/WebCore/page/Page.h:

* Source/WebCore/page/PageConfiguration.h: Add MediaSessionManagerFactory.

* Source/WebCore/page/cocoa/PageCocoa.mm:
(WebCore::Page::setPresentingApplicationBundleIdentifier): Use mediaSessionManager() instead
of PlatformMediaSessionManager directly.

* Source/WebCore/platform/RemoteCommandListener.h:
* Source/WebCore/platform/audio/MediaSessionManagerInterface.h: Inherit from
ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr.
(WebCore::MediaSessionManagerInterface::addSupportedCommand):
(WebCore::MediaSessionManagerInterface::removeSupportedCommand):
(WebCore::MediaSessionManagerInterface::supportedCommands const):

* Source/WebCore/platform/audio/PlatformMediaSession.cpp:
(WebCore::PlatformMediaSession::setActive): Use the client&apos;s mediaSessionManager instead of
PlatformMediaSessionManager::singleton.
(WebCore::PlatformMediaSession::setState): Ditto.
(WebCore::PlatformMediaSession::clientWillBeginPlayback): Ditto.
(WebCore::PlatformMediaSession::processClientWillPausePlayback): Ditto.
(WebCore::PlatformMediaSession::stopSession): Ditto.
(WebCore::PlatformMediaSession::isPlayingToWirelessPlaybackTargetChanged): Ditto.
(WebCore::PlatformMediaSession::canProduceAudioChanged): Ditto.
(WebCore::PlatformMediaSession::clientCharacteristicsChanged): Ditto.

* Source/WebCore/platform/audio/PlatformMediaSessionInterface.cpp:
* Source/WebCore/platform/audio/PlatformMediaSessionInterface.h: Move structs, classes,
and enums used by both PlatformMediaSessionInterface andn MediaSessionManagerInterface to
PlatformMediaSessionTypes.h to avoid recursive includes.
(WebCore::PlatformMediaSessionInterface::sessionManager const):

* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::create): Add optional PageIdentifier parameter.
(WebCore::PlatformMediaSessionManager::updateNowPlayingInfoIfNecessary): Make not static.
(WebCore::PlatformMediaSessionManager::updateAudioSessionCategoryIfNecessary): Ditto.
(WebCore::sharedPlatformMediaSessionManager): Deleted.
(WebCore::PlatformMediaSessionManager::singleton): Deleted.
(WebCore::PlatformMediaSessionManager::singletonIfExists): Deleted.
(WebCore::PlatformMediaSessionManager::setAlternateWebMPlayerEnabled): Deleted.
(WebCore::PlatformMediaSessionManager::alternateWebMPlayerEnabled): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
(WebCore::PlatformMediaSessionManager::ref const): Deleted.
(WebCore::PlatformMediaSessionManager::deref const): Deleted.

* Source/WebCore/platform/audio/PlatformMediaSessionTypes.h: Added.

* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::MediaSessionManagerCocoa): Don&apos;t initialize m_defaultBufferSize
as it requires a sync call to the GPU process.
(WebCore::MediaSessionManagerCocoa::updateSessionState): Initialize m_defaultBufferSize if
necessary.
(WebCore::PlatformMediaSessionManager::create): Add optional PageIdentifier parameter.

* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.mm:
(WebCore::PlatformMediaSessionManager::create): Ditto.

* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::cocoaWebMPlayerEnabled):
(WebCore::MediaPlayer::setCocoaWebMPlayerEnabled):
* Source/WebCore/platform/graphics/MediaPlayer.h:

* Source/WebCore/platform/graphics/cocoa/MediaEngineConfigurationFactoryCocoa.cpp:
(WebCore::computeMediaCapabilitiesInfo): Use the factory function to look up the correct
media session for the configuration page identifier.
* Source/WebCore/platform/mediacapabilities/MediaDecodingConfiguration.h: Add optional
page identifier.

* Source/WebCore/platform/mediacapabilities/MediaEngineConfigurationFactory.cpp:
(WebCore::mediaSessionManagerProvider):
(WebCore::MediaEngineConfigurationFactory::setMediaSessionManagerProvider):
(WebCore::MediaEngineConfigurationFactory::mediaSessionManagerForPageIdentifier):
* Source/WebCore/platform/mediacapabilities/MediaEngineConfigurationFactory.h:

* Source/WebCore/platform/mock/MockRealtimeAudioSource.cpp:
(WebCore::MockRealtimeAudioSource::startProducingData): Remove the call to
PlatformMediaSessionManager::singleton().sessionCanProduceAudioChanged, it is done in
MediaStream::startProducingData. Remove the asserts about audio session category and mode
because they will be changed asynchronously.

* Source/WebCore/testing/InternalSettings.cpp:
(WebCore::InternalSettings::Backup::Backup):
(WebCore::InternalSettings::Backup::restoreTo):
(WebCore::InternalSettingsWrapper::InternalSettingsWrapper): Deleted.
(WebCore::InternalSettingsWrapper::~InternalSettingsWrapper): Deleted.
(): Deleted.
(WebCore::InternalSettingsWrapper::internalSettings const): Deleted.
(WebCore::InternalSettings::supplementName): Deleted.
(WebCore::InternalSettings::from): Deleted.
(WebCore::InternalSettings::hostDestroyed): Deleted.
(WebCore::InternalSettings::InternalSettings): Deleted.
(WebCore::InternalSettings::create): Deleted.
(WebCore::InternalSettings::resetToConsistentState): Deleted.
(WebCore::InternalSettings::settings const): Deleted.
(WebCore::InternalSettings::setStandardFontFamily): Deleted.
(WebCore::InternalSettings::setSerifFontFamily): Deleted.
(WebCore::InternalSettings::setSansSerifFontFamily): Deleted.
(WebCore::InternalSettings::setFixedFontFamily): Deleted.
(WebCore::InternalSettings::setCursiveFontFamily): Deleted.
(WebCore::InternalSettings::setFantasyFontFamily): Deleted.
(WebCore::InternalSettings::setPictographFontFamily): Deleted.
(WebCore::InternalSettings::setTextAutosizingWindowSizeOverride): Deleted.
(WebCore::InternalSettings::setEditingBehavior): Deleted.
(WebCore::InternalSettings::setStorageBlockingPolicy): Deleted.
(WebCore::InternalSettings::setMinimumTimerInterval): Deleted.
(WebCore::InternalSettings::setTimeWithoutMouseMovementBeforeHidingControls): Deleted.
(WebCore::InternalSettings::setFontLoadTimingOverride): Deleted.
(WebCore::InternalSettings::setAllowAnimationControlsOverride): Deleted.
(WebCore::InternalSettings::setUserInterfaceDirectionPolicy): Deleted.
(WebCore::InternalSettings::setSystemLayoutDirection): Deleted.
(WebCore::InternalSettings::forcedColorsAreInvertedAccessibilityValue const): Deleted.
(WebCore::InternalSettings::setForcedColorsAreInvertedAccessibilityValue): Deleted.
(WebCore::InternalSettings::forcedDisplayIsMonochromeAccessibilityValue const): Deleted.
(WebCore::InternalSettings::setForcedDisplayIsMonochromeAccessibilityValue): Deleted.
(WebCore::InternalSettings::forcedPrefersContrastAccessibilityValue const): Deleted.
(WebCore::InternalSettings::setForcedPrefersContrastAccessibilityValue): Deleted.
(WebCore::InternalSettings::forcedPrefersReducedMotionAccessibilityValue const): Deleted.
(WebCore::InternalSettings::setForcedPrefersReducedMotionAccessibilityValue): Deleted.
(WebCore::InternalSettings::forcedSupportsHighDynamicRangeValue const): Deleted.
(WebCore::InternalSettings::setForcedSupportsHighDynamicRangeValue): Deleted.
(WebCore::InternalSettings::vp9DecoderEnabled const): Deleted.
(WebCore::InternalSettings::setCustomPasteboardDataEnabled): Deleted.
(WebCore::InternalSettings::setShouldManageAudioSessionCategory): Deleted.
(WebCore::InternalSettings::setShouldDisplayTrackKind): Deleted.
(WebCore::InternalSettings::shouldDisplayTrackKind): Deleted.
(WebCore::InternalSettings::setEditableRegionEnabled): Deleted.
(WebCore::InternalSettings::setCanStartMedia): Deleted.
(WebCore::InternalSettings::setUseDarkAppearance): Deleted.
(WebCore::InternalSettings::setUseElevatedUserInterfaceLevel): Deleted.
(WebCore::InternalSettings::setAllowUnclampedScrollPosition): Deleted.
(WebCore::InternalSettings::setShouldDeactivateAudioSession): Deleted.
(WebCore::InternalSettings::setShouldMockBoldSystemFontForAccessibility): Deleted.
(WebCore::InternalSettings::setDefaultAudioContextSampleRate): Deleted.
(WebCore::InternalSettings::setAllowedMediaContainerTypes): Deleted.
(WebCore::InternalSettings::setAllowedMediaCodecTypes): Deleted.
(WebCore::InternalSettings::setAllowedMediaVideoCodecIDs): Deleted.
(WebCore::InternalSettings::setAllowedMediaAudioCodecIDs): Deleted.
(WebCore::InternalSettings::setAllowedMediaCaptionFormatTypes): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
(WebCore::Internals::beginMediaSessionInterruption):
(WebCore::Internals::endMediaSessionInterruption):
(WebCore::Internals::applicationWillBecomeInactive):
(WebCore::Internals::applicationDidBecomeActive):
(WebCore::Internals::applicationWillEnterForeground const):
(WebCore::Internals::applicationDidEnterBackground const):
(WebCore::Internals::setMediaSessionRestrictions):
(WebCore::Internals::mediaSessionRestrictions const):
(WebCore::Internals::postRemoteControlCommand):
(WebCore::Internals::simulateSystemSleep const):
(WebCore::Internals::simulateSystemWake const):
(WebCore::Internals::nowPlayingMetadata const):
(WebCore::Internals::nowPlayingState const):
(WebCore::Internals::isMonitoringWirelessRoutes const):
(WebCore::Internals::audioCaptureSourceCount const):
(WebCore::Internals::processWillSuspend):
(WebCore::Internals::processDidResume):
(WebCore::Internals::setIsPlayingToAutomotiveHeadUnit):
(WebCore::Internals::platformSupportedCommands const):
(WebCore::Internals::sessionManager const):
* Source/WebCore/testing/Internals.h:

* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::updateGPUProcessPreferences): setAlternateWebMPlayerEnabled moved from
PlatformMediaSessionManager to MediaPlayer.

* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::processWillSuspend): Added, used to
(WebKit::WebPageProxy::processDidResume):
* Source/WebKit/UIProcess/WebPageProxy.h:

* Source/WebKit/WebProcess/GPU/GPUProcessConnection.cpp:
(WebKit::GPUProcessConnection::didReceiveRemoteCommand): Pass the remote command to the
WebProcess, which will pass it to the each WebPage, which will pass it to the Page, which
will pass it to the media session manager.

* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::requestActiveNowPlayingSessionInfo):
(WebKit::WebPage::processSystemWillSleep const):
(WebKit::WebPage::processSystemDidWake const):

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_textAnimationController):
(WebKit::WebPage::updatePreferences):
(WebKit::WebPage::processWillSuspend):
(WebKit::WebPage::processDidResume):
(WebKit::WebPage::didReceiveRemoteCommand):
(WebKit::WebPage::startObservingNowPlayingMetadata):
(WebKit::WebPage::stopObservingNowPlayingMetadata):
(WebKit::WebPage::mediaSessionManager const):
(WebKit::WebPage::mediaSessionManagerIfExists const):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::applicationWillResignActive):
(WebKit::WebPage::applicationDidEnterBackground):
(WebKit::WebPage::applicationWillEnterForeground):
(WebKit::WebPage::applicationDidBecomeActive):
(WebKit::WebPage::applicationDidEnterBackgroundForMedia):
(WebKit::WebPage::applicationWillEnterForegroundForMedia):

* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::prepareToSuspend):
(WebKit::WebProcess::processDidResume):
(WebKit::WebProcess::didReceiveRemoteCommand):
* Source/WebKit/WebProcess/WebProcess.h:

* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::systemWillSleep):
(WebKit::WebProcess::systemDidWake):

Canonical link: <a href="https://commits.webkit.org/297837@main">https://commits.webkit.org/297837@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/12cca9be7231bccce6ba58df93de6ed0e934203f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113058 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32793 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23271 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119262 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/63650 "Built successfully") | ✅ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115020 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33445 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41356 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86052 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/36712 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ✅ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116005 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26706 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101699 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66362 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25984 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19829 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63020 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96093 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19903 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122483 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40136 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29931 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94899 "Found 2 new test failures: imported/w3c/web-platform-tests/html/semantics/forms/constraints/infinite_backtracking.html inspector/dom-debugger/event-animation-frame-breakpoints.html (failure)") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40520 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97918 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94641 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24154 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39782 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17588 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36261 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40022 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45521 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39663 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42996 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41400 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Source/WebCore/Modules/mediacapabilities/MediaCapabilities.cpp:
```diff
@@ -183,10 +183,14 @@ static void gatherDecodingInfo(Document& document, MediaDecodingConfiguration&&
     configuration.canExposeVP9 = document.settings().vp9DecoderEnabled();
 #endif
 
+    RefPtr protectedPage = document.protectedPage();
```
Nit: per guidelines, this should be:
```suggestion
    RefPtr protectedPage = document.page();
```

aestes commented on Source/WebCore/Modules/mediasession/MediaSession.cpp:
```diff
@@ -261,7 +261,7 @@ void MediaSession::setPlaybackState(MediaSessionPlaybackState state)
 {
 #if ENABLE(MEDIA_STREAM)
-    RefPtr document = this->document();
+    RefPtr document = this->protectedDocument();
```
```suggestion
    RefPtr document = this->document();
```

aestes commented on Source/WebCore/Modules/mediasession/MediaSession.cpp:
```diff
@@ -270,31 +270,39 @@ ExceptionOr<void> MediaSession::setActionHandler(MediaSessionAction action, RefP
-            document->setShouldListenToVoiceActivity(!!handler);
-    }
+    if (action == MediaSessionAction::Voiceactivity)
+        document->setShouldListenToVoiceActivity(!!handler);
```
How do we know `document` is non-null here?

aestes commented on Source/WebCore/Modules/mediasession/MediaSession.cpp:
```diff
@@ -392,6 +400,24 @@ Document* MediaSession::document() const
+
+RefPtr<MediaSessionManagerInterface> MediaSession::sessionManager() const
+{
+    RefPtr document = this->protectedDocument();
```
```suggestion
    RefPtr document = this->document();
```

aestes commented on Source/WebCore/Modules/mediasession/MediaSession.cpp:
```diff
@@ -392,6 +400,24 @@ Document* MediaSession::document() const
+    if (!document)
+        return nullptr;
+
+    RefPtr page = document->protectedPage();
```
```suggestion
    RefPtr page = document->page();
```

aestes commented on Source/WebCore/Modules/mediasession/MediaSession.cpp:
```diff
@@ -452,11 +478,15 @@ void MediaSession::notifyActionHandlerObservers()
 RefPtr<HTMLMediaElement> MediaSession::activeMediaElement() const
 {
-    RefPtr document = this->document();
+    RefPtr document = this->protectedDocument();
```
```suggestion
    RefPtr document = this->document();
```

aestes commented on Source/WebCore/Modules/mediasession/MediaSession.cpp:
```diff
@@ -452,11 +478,15 @@ void MediaSession::notifyActionHandlerObservers()
         return nullptr;
 
-    return HTMLMediaElement::bestMediaElementForRemoteControls(MediaElementSession::PlaybackControlsPurpose::MediaSession, document.get());
+    RefPtr page = document->protectedPage();
```
```suggestion
    RefPtr page = document->page();
```

aestes commented on Source/WebCore/Modules/mediasession/MediaSession.cpp:
```diff
@@ -532,8 +562,8 @@ void MediaSession::updateNowPlayingInfo(NowPlayingInfo& info)
 {
-    RefPtr document = this->document();
-    if (!document->isFullyActive()) {
+    RefPtr document = this->protectedDocument();
```
```suggestion
    RefPtr document = this->document();
```

aestes commented on Source/WebCore/Modules/webaudio/BaseAudioContext.cpp:
```diff
@@ -983,6 +985,20 @@ WTFLogChannel& BaseAudioContext::logChannel() const
+    if (!document)
+        return nullptr;
+
+    RefPtr page = document->protectedPage();
```
```suggestion
    RefPtr page = document->page();
```

aestes commented on Source/WebCore/html/HTMLMediaElement.cpp:
```diff
@@ -786,37 +795,17 @@ HTMLMediaElement::~HTMLMediaElement()
     schedulePlaybackControlsManagerUpdate();
+
+    invalidateMediaSession();
+    unregisterWithDocument(Ref<Document> { document() });
```
Could this use `protectedDocument()` instead?

aestes commented on Source/WebCore/html/HTMLMediaElement.cpp:
```diff
@@ -6328,7 +6317,8 @@ bool HTMLMediaElement::couldPlayIfEnoughData() const
         return false;
 
-    if (!canProduceAudio() || PlatformMediaSessionManager::singleton().hasActiveAudioSession())
+    RefPtr manager = sessionManager();
+    if (!canProduceAudio() || (manager && manager->hasActiveAudioSession()))
```
Nit: you could avoid some refcount churn by breaking this apart (i.e. no need to create a `manager` if `!canProduceAudio()`).

aestes commented on Source/WebCore/html/HTMLMediaElement.cpp:
```diff
@@ -10240,6 +10230,14 @@ bool HTMLMediaElement::limitedMatroskaSupportEnabled() const
 
+RefPtr<MediaSessionManagerInterface> HTMLMediaElement::sessionManager() const
+{
+    if (RefPtr page = document().protectedPage())
```
```suggestion
    if (RefPtr page = document().page())
```

aestes commented on Source/WebCore/html/MediaElementSession.cpp:
```diff
@@ -663,10 +672,12 @@ bool MediaElementSession::canShowControlsManager(PlaybackControlsPurpose purpose
         return true;
     }
 
+    RefPtr manager = sessionManager();
+    bool registeredAsNowPlayingApplication = manager && manager->registeredAsNowPlayingApplication();
     if (client().presentationType() == MediaType::Audio && purpose == PlaybackControlsPurpose::NowPlaying) {
```
Nit: you could move this inside the outer `if` statement to avoid some refcount churn for non-audio, non-NowPlaying cases.

aestes commented on Source/WebCore/page/Page.cpp:
```diff
@@ -2968,9 +2970,31 @@ void Page::schedulePlaybackControlsManagerUpdate()
+RefPtr<HTMLMediaElement> Page::bestMediaElementForRemoteControls(MediaElementSession::PlaybackControlsPurpose purpose, Document* document)
+{
+    auto selectedSession = mediaSessionManager().bestEligibleSessionForRemoteControls([document] (auto& session) {
+        auto* mediaElementSession = dynamicDowncast<MediaElementSession>(session);
```
```suggestion
        RefPtr mediaElementSession = dynamicDowncast<MediaElementSession>(session);
```

aestes commented on Source/WebCore/page/Page.cpp:
```diff
@@ -2968,9 +2970,31 @@ void Page::schedulePlaybackControlsManagerUpdate()
+        if (!mediaElementSession)
+            return false;
+
+        RefPtr element = mediaElementSession->protectedElement();
```
```suggestion
        RefPtr element = mediaElementSession->element();
```

aestes commented on Source/WebCore/testing/InternalSettings.cpp:
```diff
@@ -65,10 +65,11 @@ InternalSettings::Backup::Backup(Settings& settings)
+{
 #if ENABLE(VIDEO) || ENABLE(WEB_AUDIO)
-    , m_shouldDeactivateAudioSession(PlatformMediaSessionManager::shouldDeactivateAudioSession())
+    if (RefPtr page = settings.protectedPage())
```
```suggestion
    if (RefPtr page = settings.page())
```

aestes commented on Source/WebCore/testing/InternalSettings.cpp:
```diff
@@ -120,7 +121,8 @@ void InternalSettings::Backup::restoreTo(Settings& settings)
 
 #if ENABLE(VIDEO) || ENABLE(WEB_AUDIO)
-    PlatformMediaSessionManager::setShouldDeactivateAudioSession(m_shouldDeactivateAudioSession);
+    if (RefPtr page = settings.protectedPage())
```
```suggestion
    if (RefPtr page = settings.page())
```

aestes commented on Source/WebCore/testing/Internals.cpp:
```diff
@@ -7895,6 +7943,19 @@ ExceptionOr<Vector<Internals::FrameDamage>> Internals::getFrameDamageHistory() c
+    if (!document)
+        return nullptr;
+
+    RefPtr page = document->protectedPage();
```
```suggestion
    RefPtr page = document->page();
```

aestes commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -10086,15 +10106,16 @@ void WebPage::setDefaultSpatialTrackingLabel(const String& label)
     });
 
-    WebCore::PlatformMediaSessionManager::singleton().addNowPlayingMetadataObserver(*m_nowPlayingMetadataObserver);
+sessionManager->addNowPlayingMetadataObserver(*m_nowPlayingMetadataObserver);
```
```suggestion
    sessionManager->addNowPlayingMetadataObserver(*m_nowPlayingMetadataObserver);
```

aestes commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -10419,6 +10441,20 @@ void WebPage::frameViewLayoutOrVisualViewportChanged(const LocalFrameView& frame
 
+RefPtr<MediaSessionManagerInterface> WebPage::mediaSessionManager() const
+{
+    RefPtr page { protectedCorePage() };
```
```suggestion
    RefPtr page { corePage() };
```

aestes commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -10419,6 +10441,20 @@ void WebPage::frameViewLayoutOrVisualViewportChanged(const LocalFrameView& frame
+
+MediaSessionManagerInterface* WebPage::mediaSessionManagerIfExists() const
+{
+    RefPtr page { protectedCorePage() };
```
```suggestion
    RefPtr page { corePage() };
```

eric-carlson committed:
Remove PlatformMediaSessionManager::singleton
https://bugs.webkit.org/show_bug.cgi?id=292687
rdar://150885431

Reviewed by Andy Estes.

For site-isolation we will need to move MediaSessionManager and related classes to the
GPU process and have a different MediaSessionManagerInterface proxy for every Page. As a
step in this direction, get rid of PlatformMediaSessionManager::singleton and add
Page::mediaSessionManager and remove all PlatformMediaSessionManager static methods.

Add a factory function to PageConfiguration that WebProcess will use to create a
RemoveMediaSessionManager once it exists.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/audiosession/DOMAudioSession.cpp:
(WebCore::DOMAudioSession::setType): Use Page::mediaSessionManager instead of
PlatformMediaSessionManager::singleton.

* Source/WebCore/Modules/mediacapabilities/MediaCapabilities.cpp:
(WebCore::gatherDecodingInfo): Ditto.

* Source/WebCore/Modules/mediasession/MediaSession.cpp:
(WebCore::MediaSession::setActionHandler): Ditto.
(WebCore::MediaSession::protectedDocument const): Add Document RefPtr
(WebCore::MediaSession::sessionManager const): Add accessor.
(WebCore::MediaSession::activeMediaElement const):
(WebCore::MediaSession::updateNowPlayingInfo):
(WebCore::MediaSession::updateCaptureState):
(WebCore::MediaSession::document const): Deleted.
* Source/WebCore/Modules/mediasession/MediaSession.h:

* Source/WebCore/Modules/mediastream/MediaStream.cpp:
(WebCore::MediaStream::startProducingData): Call `sessionCanProduceAudioChanged` if the
stream has an audio track so we don't need to do call it from the Mock audio source.
(WebCore::MediaStream::mediaSessionManager const):
* Source/WebCore/Modules/mediastream/MediaStream.h:

* Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp:
(WebCore::MediaStreamTrack::MediaStreamTrack): Use Page::mediaSessionManager instead of
PlatformMediaSessionManager::singleton.
(WebCore::MediaStreamTrack::~MediaStreamTrack): Ditto.
(WebCore::MediaStreamTrack::stopTrack): Ditto.
(WebCore::MediaStreamTrack::trackEnded): Ditto.
(WebCore::MediaStreamTrack::trackMutedChanged): Ditto.
(WebCore::MediaStreamTrack::mediaSessionManager const):
* Source/WebCore/Modules/mediastream/MediaStreamTrack.h:

* Source/WebCore/Modules/mediastream/UserMediaRequest.cpp:
(WebCore::UserMediaRequest::start): Ditto.

* Source/WebCore/Modules/webaudio/AudioContext.cpp:
(WebCore::AudioContext::defaultDestinationWillBecomeConnected): Ditto.
(WebCore::AudioContext::sessionManager const): Ditto.
* Source/WebCore/Modules/webaudio/AudioContext.h:

* Source/WebCore/Modules/webaudio/BaseAudioContext.cpp:
(WebCore::BaseAudioContext::setState): Ditto.
(WebCore::BaseAudioContext::mediaSessionManager const):
* Source/WebCore/Modules/webaudio/BaseAudioContext.h:

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::visibilityStateChanged): Ditto.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::bestMediaElementForRemoteControls): Deleted.
(WebCore::HTMLMediaElement::seekToPlaybackPositionEndedTimerFired): Use
Page::mediaSessionManager instead of PlatformMediaSessionManager::singleton.
(WebCore::HTMLMediaElement::couldPlayIfEnoughData const): Ditto.
(WebCore::HTMLMediaElement::shouldDisableSleep const): Ditto.
(WebCore::HTMLMediaElement::shouldOverrideBackgroundPlaybackRestriction const): Ditto.
(WebCore::HTMLMediaElement::sessionManager const):
* Source/WebCore/html/HTMLMediaElement.h:

* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::MediaElementSession::isVisibleInViewportChanged): Use
PlatformMediaSessionClient::sessionManager instead of PlatformMediaSessionManager::singleton.
(WebCore::MediaElementSession::clientDataBufferingTimerFired): Ditto.
(WebCore::MediaElementSession::updateClientDataBuffering): Ditto.
(WebCore::MediaElementSession::canShowControlsManager const): Ditto.
(WebCore::MediaElementSession::setHasPlaybackTargetAvailabilityListeners): Ditto.

* Source/WebCore/page/Page.cpp:
(WebCore::Page::bestMediaElementForRemoteControls): Move from HTMLMediaElement.
(WebCore::Page::mediaPlaybackExists): Use mediaSessionManager instance variable instead of
PlatformMediaSessionManager::singletonIfExists.
(WebCore::Page::mediaPlaybackIsPaused): Ditto.
(WebCore::Page::pauseAllMediaPlayback): Ditto.
(WebCore::Page::suspendAllMediaPlayback): Ditto.
(WebCore::Page::resumeAllMediaPlayback): Ditto.
(WebCore::Page::suspendAllMediaBuffering): Ditto.
(WebCore::Page::resumeAllMediaBuffering): Ditto.
(WebCore::Page::setActivityState): Ditto.
(WebCore::Page::activeNowPlayingSessionUpdateTimerFired): Ditto.
(WebCore::Page::setPresentingApplicationAuditToken): Ditto.
(WebCore::Page::mediaSessionManager): Allocate `m_mediaSessionManager` with the factory
function if it doesn't exist. If m_mediaSessionManagerFactory is null, true for WK1 and
WK2 until we move the session manager to the GPU process, create a factory that allocates
and instance of PlatformMediaSessionManager.
(WebCore::Page::mediaSessionManagerIfExists const):
(WebCore::Page::mediaSessionManagerForPageIdentifier):
* Source/WebCore/page/Page.h:

* Source/WebCore/page/PageConfiguration.h: Add MediaSessionManagerFactory.

* Source/WebCore/page/cocoa/PageCocoa.mm:
(WebCore::Page::setPresentingApplicationBundleIdentifier): Use mediaSessionManager() instead
of PlatformMediaSessionManager directly.

* Source/WebCore/platform/RemoteCommandListener.h:
* Source/WebCore/platform/audio/MediaSessionManagerInterface.h: Inherit from
ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr.
(WebCore::MediaSessionManagerInterface::addSupportedCommand):
(WebCore::MediaSessionManagerInterface::removeSupportedCommand):
(WebCore::MediaSessionManagerInterface::supportedCommands const):

* Source/WebCore/platform/audio/PlatformMediaSession.cpp:
(WebCore::PlatformMediaSession::setActive): Use the client's mediaSessionManager instead of
PlatformMediaSessionManager::singleton.
(WebCore::PlatformMediaSession::setState): Ditto.
(WebCore::PlatformMediaSession::clientWillBeginPlayback): Ditto.
(WebCore::PlatformMediaSession::processClientWillPausePlayback): Ditto.
(WebCore::PlatformMediaSession::stopSession): Ditto.
(WebCore::PlatformMediaSession::isPlayingToWirelessPlaybackTargetChanged): Ditto.
(WebCore::PlatformMediaSession::canProduceAudioChanged): Ditto.
(WebCore::PlatformMediaSession::clientCharacteristicsChanged): Ditto.

* Source/WebCore/platform/audio/PlatformMediaSessionInterface.cpp:
* Source/WebCore/platform/audio/PlatformMediaSessionInterface.h: Move structs, classes,
and enums used by both PlatformMediaSessionInterface andn MediaSessionManagerInterface to
PlatformMediaSessionTypes.h to avoid recursive includes.
(WebCore::PlatformMediaSessionInterface::sessionManager const):

* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::create): Add optional PageIdentifier parameter.
(WebCore::PlatformMediaSessionManager::updateNowPlayingInfoIfNecessary): Make not static.
(WebCore::PlatformMediaSessionManager::updateAudioSessionCategoryIfNecessary): Ditto.
(WebCore::sharedPlatformMediaSessionManager): Deleted.
(WebCore::PlatformMediaSessionManager::singleton): Deleted.
(WebCore::PlatformMediaSessionManager::singletonIfExists): Deleted.
(WebCore::PlatformMediaSessionManager::setAlternateWebMPlayerEnabled): Deleted.
(WebCore::PlatformMediaSessionManager::alternateWebMPlayerEnabled): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
(WebCore::PlatformMediaSessionManager::ref const): Deleted.
(WebCore::PlatformMediaSessionManager::deref const): Deleted.

* Source/WebCore/platform/audio/PlatformMediaSessionTypes.h: Added.

* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::MediaSessionManagerCocoa): Don't initialize m_defaultBufferSize
as it requires a sync call to the GPU process.
(WebCore::MediaSessionManagerCocoa::updateSessionState): Initialize m_defaultBufferSize if
necessary.
(WebCore::PlatformMediaSessionManager::create): Add optional PageIdentifier parameter.

* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.mm:
(WebCore::PlatformMediaSessionManager::create): Ditto.

* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::cocoaWebMPlayerEnabled):
(WebCore::MediaPlayer::setCocoaWebMPlayerEnabled):
* Source/WebCore/platform/graphics/MediaPlayer.h:

* Source/WebCore/platform/graphics/cocoa/MediaEngineConfigurationFactoryCocoa.cpp:
(WebCore::computeMediaCapabilitiesInfo): Use the factory function to look up the correct
media session for the configuration page identifier.
* Source/WebCore/platform/mediacapabilities/MediaDecodingConfiguration.h: Add optional
page identifier.

* Source/WebCore/platform/mediacapabilities/MediaEngineConfigurationFactory.cpp:
(WebCore::mediaSessionManagerProvider):
(WebCore::MediaEngineConfigurationFactory::setMediaSessionManagerProvider):
(WebCore::MediaEngineConfigurationFactory::mediaSessionManagerForPageIdentifier):
* Source/WebCore/platform/mediacapabilities/MediaEngineConfigurationFactory.h:

* Source/WebCore/platform/mock/MockRealtimeAudioSource.cpp:
(WebCore::MockRealtimeAudioSource::startProducingData): Remove the call to
PlatformMediaSessionManager::singleton().sessionCanProduceAudioChanged, it is done in
MediaStream::startProducingData. Remove the asserts about audio session category and mode
because they will be changed asynchronously.

* Source/WebCore/testing/InternalSettings.cpp:
(WebCore::InternalSettings::Backup::Backup):
(WebCore::InternalSettings::Backup::restoreTo):
(WebCore::InternalSettingsWrapper::InternalSettingsWrapper): Deleted.
(WebCore::InternalSettingsWrapper::~InternalSettingsWrapper): Deleted.
(): Deleted.
(WebCore::InternalSettingsWrapper::internalSettings const): Deleted.
(WebCore::InternalSettings::supplementName): Deleted.
(WebCore::InternalSettings::from): Deleted.
(WebCore::InternalSettings::hostDestroyed): Deleted.
(WebCore::InternalSettings::InternalSettings): Deleted.
(WebCore::InternalSettings::create): Deleted.
(WebCore::InternalSettings::resetToConsistentState): Deleted.
(WebCore::InternalSettings::settings const): Deleted.
(WebCore::InternalSettings::setStandardFontFamily): Deleted.
(WebCore::InternalSettings::setSerifFontFamily): Deleted.
(WebCore::InternalSettings::setSansSerifFontFamily): Deleted.
(WebCore::InternalSettings::setFixedFontFamily): Deleted.
(WebCore::InternalSettings::setCursiveFontFamily): Deleted.
(WebCore::InternalSettings::setFantasyFontFamily): Deleted.
(WebCore::InternalSettings::setPictographFontFamily): Deleted.
(WebCore::InternalSettings::setTextAutosizingWindowSizeOverride): Deleted.
(WebCore::InternalSettings::setEditingBehavior): Deleted.
(WebCore::InternalSettings::setStorageBlockingPolicy): Deleted.
(WebCore::InternalSettings::setMinimumTimerInterval): Deleted.
(WebCore::InternalSettings::setTimeWithoutMouseMovementBeforeHidingControls): Deleted.
(WebCore::InternalSettings::setFontLoadTimingOverride): Deleted.
(WebCore::InternalSettings::setAllowAnimationControlsOverride): Deleted.
(WebCore::InternalSettings::setUserInterfaceDirectionPolicy): Deleted.
(WebCore::InternalSettings::setSystemLayoutDirection): Deleted.
(WebCore::InternalSettings::forcedColorsAreInvertedAccessibilityValue const): Deleted.
(WebCore::InternalSettings::setForcedColorsAreInvertedAccessibilityValue): Deleted.
(WebCore::InternalSettings::forcedDisplayIsMonochromeAccessibilityValue const): Deleted.
(WebCore::InternalSettings::setForcedDisplayIsMonochromeAccessibilityValue): Deleted.
(WebCore::InternalSettings::forcedPrefersContrastAccessibilityValue const): Deleted.
(WebCore::InternalSettings::setForcedPrefersContrastAccessibilityValue): Deleted.
(WebCore::InternalSettings::forcedPrefersReducedMotionAccessibilityValue const): Deleted.
(WebCore::InternalSettings::setForcedPrefersReducedMotionAccessibilityValue): Deleted.
(WebCore::InternalSettings::forcedSupportsHighDynamicRangeValue const): Deleted.
(WebCore::InternalSettings::setForcedSupportsHighDynamicRangeValue): Deleted.
(WebCore::InternalSettings::vp9DecoderEnabled const): Deleted.
(WebCore::InternalSettings::setCustomPasteboardDataEnabled): Deleted.
(WebCore::InternalSettings::setShouldManageAudioSessionCategory): Deleted.
(WebCore::InternalSettings::setShouldDisplayTrackKind): Deleted.
(WebCore::InternalSettings::shouldDisplayTrackKind): Deleted.
(WebCore::InternalSettings::setEditableRegionEnabled): Deleted.
(WebCore::InternalSettings::setCanStartMedia): Deleted.
(WebCore::InternalSettings::setUseDarkAppearance): Deleted.
(WebCore::InternalSettings::setUseElevatedUserInterfaceLevel): Deleted.
(WebCore::InternalSettings::setAllowUnclampedScrollPosition): Deleted.
(WebCore::InternalSettings::setShouldDeactivateAudioSession): Deleted.
(WebCore::InternalSettings::setShouldMockBoldSystemFontForAccessibility): Deleted.
(WebCore::InternalSettings::setDefaultAudioContextSampleRate): Deleted.
(WebCore::InternalSettings::setAllowedMediaContainerTypes): Deleted.
(WebCore::InternalSettings::setAllowedMediaCodecTypes): Deleted.
(WebCore::InternalSettings::setAllowedMediaVideoCodecIDs): Deleted.
(WebCore::InternalSettings::setAllowedMediaAudioCodecIDs): Deleted.
(WebCore::InternalSettings::setAllowedMediaCaptionFormatTypes): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
(WebCore::Internals::beginMediaSessionInterruption):
(WebCore::Internals::endMediaSessionInterruption):
(WebCore::Internals::applicationWillBecomeInactive):
(WebCore::Internals::applicationDidBecomeActive):
(WebCore::Internals::applicationWillEnterForeground const):
(WebCore::Internals::applicationDidEnterBackground const):
(WebCore::Internals::setMediaSessionRestrictions):
(WebCore::Internals::mediaSessionRestrictions const):
(WebCore::Internals::postRemoteControlCommand):
(WebCore::Internals::simulateSystemSleep const):
(WebCore::Internals::simulateSystemWake const):
(WebCore::Internals::nowPlayingMetadata const):
(WebCore::Internals::nowPlayingState const):
(WebCore::Internals::isMonitoringWirelessRoutes const):
(WebCore::Internals::audioCaptureSourceCount const):
(WebCore::Internals::processWillSuspend):
(WebCore::Internals::processDidResume):
(WebCore::Internals::setIsPlayingToAutomotiveHeadUnit):
(WebCore::Internals::platformSupportedCommands const):
(WebCore::Internals::sessionManager const):
* Source/WebCore/testing/Internals.h:

* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::updateGPUProcessPreferences): setAlternateWebMPlayerEnabled moved from
PlatformMediaSessionManager to MediaPlayer.

* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::processWillSuspend): Added, used to
(WebKit::WebPageProxy::processDidResume):
* Source/WebKit/UIProcess/WebPageProxy.h:

* Source/WebKit/WebProcess/GPU/GPUProcessConnection.cpp:
(WebKit::GPUProcessConnection::didReceiveRemoteCommand): Pass the remote command to the
WebProcess, which will pass it to the each WebPage, which will pass it to the Page, which
will pass it to the media session manager.

* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::requestActiveNowPlayingSessionInfo):
(WebKit::WebPage::processSystemWillSleep const):
(WebKit::WebPage::processSystemDidWake const):

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_textAnimationController):
(WebKit::WebPage::updatePreferences):
(WebKit::WebPage::processWillSuspend):
(WebKit::WebPage::processDidResume):
(WebKit::WebPage::didReceiveRemoteCommand):
(WebKit::WebPage::startObservingNowPlayingMetadata):
(WebKit::WebPage::stopObservingNowPlayingMetadata):
(WebKit::WebPage::mediaSessionManager const):
(WebKit::WebPage::mediaSessionManagerIfExists const):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::applicationWillResignActive):
(WebKit::WebPage::applicationDidEnterBackground):
(WebKit::WebPage::applicationWillEnterForeground):
(WebKit::WebPage::applicationDidBecomeActive):
(WebKit::WebPage::applicationDidEnterBackgroundForMedia):
(WebKit::WebPage::applicationWillEnterForegroundForMedia):

* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::prepareToSuspend):
(WebKit::WebProcess::processDidResume):
(WebKit::WebProcess::didReceiveRemoteCommand):
* Source/WebKit/WebProcess/WebProcess.h:

* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::systemWillSleep):
(WebKit::WebProcess::systemDidWake):

Canonical link: https://commits.webkit.org/297837@main
